File: androidsupport/src/main/java/com/jarklee/androidsupport/callback/LifeCycleListener.java
/*
 * ******************************************************************************
 *  Copyright Ⓒ 2017. TrinhQuan. All right reserved
 *  Author: TrinhQuan. Created on 2017/1/6
 *  Contact: trinhquan.171093@gmail.com
 * ******************************************************************************
 */

package com.jarklee.androidsupport.callback;

import com.jarklee.androidsupport.lifecycle.LifeCycleHook;

public interface LifeCycleListener {

    void onReady(LifeCycleHook hook);

    void onCreate();

    void onStart();

    void onResume();

    void onPause();

    void onStop();

    void onDestroy();
}

File: androidsupport-kotlin/src/main/java/com/jarklee/androidsupport/callback/LifeCycleListener.kt
/*
 * ******************************************************************************
 *  Copyright Ⓒ 2016. TrinhQuan. All right reserved
 *  Author: TrinhQuan. Created on 2016/12/20
 *  Contact: trinhquan.171093@gmail.com
 * ******************************************************************************
 */

package com.jarklee.androidsupport.callback

import com.jarklee.androidsupport.lifecycle.LifeCycleHook

interface LifeCycleListener {

    fun onReady(hook: LifeCycleHook)

    fun onCreate()

    fun onStart()

    fun onResume()

    fun onPause()

    fun onStop()

    fun onDestroy()
}
